packages/docs-reanimated/docs/debugging/logger-configuration.mdx

```diff
@@ -0,0 +1,67 @@
+---
+id: logger-configuration
+sidebar_label: Logger configuration
+sidebar_position: 2
+---
+
+# Logger configuration
+
+Reanimated shows warnings that indicate misuses of the library API, such as modifying the [shared value](/docs/fundamentals/glossary#shared-value) during component re-render. These logs can be configured to be more or less verbose. The **default** configuration shows **all warnings and errors**. You can change this behavior by using the `configureReanimatedLogger` function.
+
+## Reference
+
+To configure the Reanimated logger, import `configureReanimatedLogger` from `react-native-reanimated` and call it with the desired configuration.
+
+```js
+import {
+  configureReanimatedLogger,
+  ReanimatedLogLevel,
+} from 'react-native-reanimated';
+
+// This is a default configuration
+configureReanimatedLogger({
+  level: ReanimatedLogLevel.warn,
+  strict: true, // Reanimated runs in strict mode by default
+});
+```
+
+<details>
+<summary>Type definitions</summary>
+
+```typescript
+function configureReanimatedLogger(config: LoggerConfig): void;
+
+type LoggerConfig = {
+  level?: ReanimatedLogLevel;
+  strict?: boolean;
+};
+
+enum ReanimatedLogLevel {
+  warn = 1,
+  error = 2,
+}
+```
+
+</details>
+
+### Configuration options
+
+#### `level`
+
+A value od the `ReanimatedLogLevel` enum that defines the minimum level of the logs that will be shown.
+
+#### `strict`
+
+A boolean value that enables or disables strict mode. When strict mode is enabled, Reanimated will show more warnings that can help you to catch potential issues in your code.
+
+## Remarks
+
+- The logger configuration is global and affects all Reanimated animations in your app.
```

Comment:
the docs should say whether the logger is enabled in production build

Reply:
It depends. The logger didn't change anything compared to the previous behavior that we had while using `console.warn`/`console.error`. Some warnings are shown only in `__DEV__` and some not.

I don't want to write that the logger will be included in production builds as it will include most of the logs but not all of them.

I think we should add a possibility to configure whether we want to see warnings in prod as well via the logger config in the future.

What do you suggest writing for now?